### ISSUE TITLE
Input shape for an RNN

### ISSUE THREAD

**Issue by DarylWM** [tflearn/tflearn#68]
I've looked at the LSTM example, and other examples I could find, but I'm still unclear how to shape the data for my example. My training data set comprises a number of sequences of 30 float values; the target value for each sequence is a single float value. Given a sequence of 30 values, I would like to use the model to predict the target value.

What would you suggest for the input shape?

**Comment by aymericdamien** [comment]
I think it should be: input_data: (None, 30, 1) and targets: (None, 1). You can use mean_square loss.

**Comment by DarylWM** [comment]
Thanks. So if I had a training set of 200 sequences, each sequence having 30 values, it would be shape=[None, 30, 200]?

Regarding the output shape, how do I specify the model to have a single output value? Does TFLearn infer the output shape from the trainY parameter in the fit() method?

**Comment by aymericdamien** [comment]
That should be (200, 30, 1), the first dimension is your batch size. Actually that is why input_data(shape=(none, 30, 1), because you can change the batch size, so we usually set its dimension to 'None' in TensorFlow.
You do not need to specify an output shape, TFLearn will figure it out alone.

**Comment by DarylWM** [comment]
Thanks - that's very helpful.
